backends/bmv2/simple_switch/simpleSwitch.cpp

```diff
@@ -154,6 +154,19 @@ Util::IJson* ExternConverter_clone3::convertExternFunction(
         modelError("Expected 3 arguments for %1%", mc);
         return nullptr;
     }
+    auto exp = mc->arguments->at(2)->expression;
+    if (exp->type->is<IR::Type_Struct>()) {
+        auto strct = exp->type->to<IR::Type_Struct>();
+        for (auto f : strct->fields) {
+            auto ftype = f->type;
+            if (ftype->is<IR::Type_Header>()) {
+                ::error("v1 model does not support header %1% inside struct for clone3 arg %2%",
+                        f, exp);
+                return nullptr;
+            }
+        }
+    }
+
```

Comment:
The same restriction applies to resubmit and recirculate field lists, too.  Would it be as straightforward to add such a check for them, too?

Reply:
PSA does clone, recirculate, and resubmit preservation of selected metadata fields in a different way than the current implementation does, and it isn't currently working yet anyway, so nothing working there yet to update.